app/models/locale_site_setting.rb

```diff
@@ -8,8 +8,20 @@ def self.valid_value?(val)
   def self.values
     @values ||=
```

Comment:
Memoization might be a problem here when used on a multisite. We are moving the localized names into a locale file which allows every site owner to customize that name. We could prevent that from happening, but why introduce a special case? At a quick glance, it looks like `LocaleSiteSetting.values` isn't in a hot path. We could calculate the `name` on demand. That way we'd always use the translations of the current site and it would mean you could skip the complexity of `i18n_dependent_choices`.

Reply:
> At a quick glance, it looks like LocaleSiteSetting.values isn't in a hot path

What do you mean by "hot path"?

Reply:
`LocaleSiteSetting.values` is evaluated before the `I18n` files are available, so even without the multisite memoization issue, the list of values would be

```
[
    {
        "name": null,
        "value": "ar"
    },
```

instead of what we want 

```
[
    {
        "name": "Arabic (اللغة العربية)",
        "value": "ar"
    },
```